wp-includes/class.wp-service-workers.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Dependencies API: WP_Service_Workers class
+ *
+ * @since ?
+ *
+ * @package PWA
+ */
+
+/**
+ * Class used to register service workers.
+ *
+ * @since ?
+ *
+ * @see WP_Dependencies
+ * @todo Does it make sense to extend WP_Scripts, maybe WP_Dependencies is enough?
+ */
+class WP_Service_Workers extends WP_Scripts {
+
+	/**
+	 * Param for service workers.
+	 *
+	 * @var string
+	 */
+	public $query_var = 'wp_service_worker';
+
+	/**
+	 * Array of scopes.
+	 *
+	 * @var array
+	 */
+	public $scopes = array();
```

Comment:
Is this `$scopes` needed? There is already `scope` data attached to each registered service worker, so then when getting the scripts to print when rendering the service worker it can just iterate over all the scripts and gather up the matching scope. If there are none matching, then return nothing with a 404.

By eliminating the var, then we don't have to keep track of the scopes in two different places, such as when a service worker script is unregistered.

Reply:
True. I was originally thinking of using only `$scopes` instead of using handles, however, this way we wouldn't be able to leverage the existing `do_items()` method which takes care of the `$deps` as well.